src/main/java/com/ericsson/ei/subscription/SubscriptionHandler.java

```diff
@@ -73,12 +74,18 @@
      */
     @Async("subscriptionHandlerExecutor")
     public void checkSubscriptionForObject(final String aggregatedObject,
-            final String id) {
+            final String id) {    
+    	LOGGER.info("**** Thread name {} starts for {} **** ", Thread.currentThread().getName(), id);
+    	long start = System.currentTimeMillis();
         final List<String> subscriptions = mongoDBHandler.getAllDocuments(
                 subscriptionDataBaseName, subscriptionCollectionName);
+        long end = System.currentTimeMillis();
+        long elapsedTime = end - start;
+        LOGGER.info("##### Response time to getAllDocuemnts in subscription handler: " + elapsedTime);       
```

Comment:
You wrote something about "subscription handler pool overflows ..." and lost events in the description,, if you remove this thread pool then the event will stay in the message-bus queue until there is free threads in main thread pool and when one thread picks up one event, that thread will process the whole EI flow of one event. Then you will no queue up events in the the Java memory that might be lost due to thread pool is full or what causing the lost of events.

Reply:
The rejection handler will fix the issue of lost events. We will see about the performance...

Reply:
Ok, where is the rejection handler code? I guess its some memory queue or something or waiting in the previous thread or something.

Nothing is more safer than keeping the event object in AMQP persistent queue until a new thread is available to process the event.